### PR TITLE
kv: remove `DelOption` method

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
`DelOption` is not very commonly used. The only one usage is `DelOption(CollectRuntimeStats)`. We can use `SetOption(CollectRuntimeStats, nil)` as an alternative. In this way, we can make interface more tidy.

Part of https://github.com/pingcap/tidb/issues/22513

### What is changed and how it works?
What's Changed:
- remove `DelOption`
- use `SetOption` instead

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

**Review by sticnarf — APPROVED** [review]

File: kv/mock_test.go [hunk 1]
```diff
@@ -44,7 +44,6 @@ func (s testMockSuite) TestInterface(c *C) {
 		mock.GetOption(23)
 	}
 	transaction.StartTS()
-	transaction.DelOption(23)
```
Comment (AndreMouche) [inline kv/mock_test.go:47]: why remove it?

Comment (disksing) [reply kv/mock_test.go:47]: don't have `DelOption` now. And following code does not rely on it.

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**